lib/model/list_model.dart

```diff
@@ -75,14 +76,7 @@ class ListModel {
 
   Future<void> remove(Item item) async {
     if (items.remove(item)) {
-      final itemsOrderedAfterRemovedItem =
-          await items.filter().orderGreaterThan(item.order).findAll();
-      for (final item in itemsOrderedAfterRemovedItem) {
-        --item.order;
-      }
-
```

Comment:
There's no need for this extra logic. `order` can be any integer. No need to constrain it to be less than the length of the list.